dagmc/dagnav.py

```diff
@@ -218,11 +218,13 @@ def id(self) -> int:
     @id.setter
     def id(self, i: int):
         """Set the DAGMC set's ID."""
-        if i in self.model.used_ids[type(self)]:
+        if i is None:
+            i = max(self.model.used_ids[type(self)], default=0) + 1
```

Comment:
I'm good w/ this as-is, but would it perhaps be more ideal to use the next available ID in the sequence (4, if the current ID space is {1-3, 5-100} for example) rather than the max of the current space +1?

Reply:
Thinking about this more, that's probably not super intuitive and only really relevant if ID spaces are flirting with the size of C++ `int` -- godspeed to anyone handling a model with that many surfaces/volumes.

Reply:
It's also a lot more complicated to keep track of robustly